module/programs/helper.js

```diff
@@ -438,4 +439,274 @@ module.exports = class ProgramsHelper {
+            };
+            let copiedQuestionsetId;
+    
+            function copyQuestionSetCallback(err, data) {
+              console.log("response from copy set", data.body, copyReq)
+              if (err || data.statusCode != 200) {
+                return resolve({
+                  message: messageConstants.apiResponses.QUESTIONSET_NOT_FOUND,
+                  status: httpStatusCode.bad_request.status,
+                })
+              } else if (data.statusCode == 200) {
+                let response = data.body
+                copiedQuestionsetId = response.result.node_id[questionSetId]
+                let url = kpServiceUrl + messageConstants.endpoints.READ_QUESTION_SET + "/" + copiedQuestionsetId + "?mode=edit";
+                request.get(url, { headers: headers }, readQuestionSetCallBack)
+              }
+              // return resolve(data.body);
+            }
+            var readQuestionSetRes = {};
+            function readQuestionSetCallBack(err, data) {
+              console.log("read copied question set", data.body)
+              if (err || data.statusCode != 200) {
+                return resolve({
+                  message: messageConstants.apiResponses.QUESTIONSET_NOT_FOUND,
+                  status: httpStatusCode.bad_request.status,
+                })
+              } else if (data.statusCode == 200) {
+                let readRes = JSON.parse(data.body)
+                readQuestionSetRes = readRes.result.questionSet
+                let childerNodes = {
+                  [readQuestionSetRes.identifier]: {
+                    metadata: {
+                      visibility: "Private"
+                    }
+                  }
+                }
+                // readQuestionSetRes.childNodes.forEach(child =>{
+                //     childerNodes[child] = {
+                //       metadata: {
+                //         visibility: "Private",
+                //       },
+                //       "objectType": "Question"
+                //     }
+                // })
+                let req = {
+                  request: {
+                    data: {
+                      nodesModified: childerNodes,
+                      hierarchy: {
+                        [readQuestionSetRes.identifier]: {
+                          name: readQuestionSetRes.name,
+                          children: readQuestionSetRes.childNodes,
+                          root: true
+                        }
+                      }
+                    }
+                  }
+                }
+                console.log("ERR_UPDATE_QS_HIERARCHY", JSON.stringify(req))
+                let updateUrl = kpServiceUrl + messageConstants.endpoints.UPDATE_QUESTION_SET_HIERARCHY;
+                request.patch(updateUrl, { headers: headers, json: true, json: req }, updateQuestionSetHierarchyCallBack)
+              }
+            }
+    
+            function updateQuestionSetHierarchyCallBack(err, data) {
```

Comment:
@aishwa8141 make all these as different functions within this helper itself and not function inside a function.